crates/common_utils/src/id_type/global_id.rs

```diff
@@ -32,8 +32,8 @@ impl GlobalEntity {
     }
 }
 
-#[derive(Debug, Clone, Hash, PartialEq, Eq)]
-pub(crate) struct CellId(LengthId<CELL_IDENTIFIER_LENGTH, CELL_IDENTIFIER_LENGTH>);
+#[derive(Debug, Clone, Hash, PartialEq, Eq, serde::Deserialize)]
```

Comment:
do not derive Deserialization directly, implement custom deserialization which also validates the constraints. You can refer to the Deser implementation of `AlphanumericId`